releasenotes/notes/pymongo-check-pin-c3cb9d1fbdc82fb3.yaml

```
@@ -0,0 +1,4 @@
+---
+fixes:
+  - |
+    pymongo: add type checking to solve an issue where ``NoneType`` instead of expected ``Pin`` object would throw an error in ``TracedTopology`` method.
```

Same comment as the 2.15 backport
```suggestion
    pymongo: Adds type checking to solve an issue where ``NoneType`` instead of expected ``Pin`` object would throw an error in ``TracedTopology`` method.
```